clang/lib/Sema/SemaType.cpp

```diff
@@ -6573,8 +6574,15 @@ static void HandleAddressSpaceTypeAttribute(QualType &Type,
     if (S.getLangOpts().HLSL)
       ASIdx = Attr.asHLSLLangAS();
 
-    if (ASIdx == LangAS::Default)
-      llvm_unreachable("Invalid address space");
+    if (ASIdx == LangAS::Default) {
+      assert(S.getLangOpts().SYCLIsDevice || S.getLangOpts().HLSL);
```

Comment:
https://llvm.org/docs/ProgrammersManual.html#programmatic-errors

```suggestion
      assert((S.getLangOpts().SYCLIsDevice || S.getLangOpts().HLSL) && "Unexpected language mode");
```